apps/nextjs/tests-e2e/tests/aila-chat/full-romans.test.ts

```diff
@@ -18,73 +18,75 @@ import {
-  // Remove extra retries when it becomes more stable
-  if (process.env.CI === "true") {
-    test.describe.configure({ retries: 4 });
-  }
-
-  test(
-    "Full aila flow with Romans fixture",
-    { tag: "@common-auth" },
-    async ({ page }) => {
-      const generationTimeout = FIXTURE_MODE === "record" ? 75000 : 50000;
-      test.setTimeout(generationTimeout * 5);
-
-      await test.step("Setup", async () => {
-        await bypassVercelProtection(page);
-        await setupClerkTestingToken({ page });
-
-        await page.goto(`${TEST_BASE_URL}/aila`);
-        await expect(page.getByTestId("chat-h1")).toBeInViewport();
-      });
-
-      const { setFixture } = await applyLlmFixtures(page, FIXTURE_MODE);
-
-      await test.step("Fill in the chat box", async () => {
-        const textbox = page.getByTestId("chat-input");
-        const sendMessage = page.getByTestId("send-message");
-        const message =
-          "Create a KS1 lesson on the end of Roman Britain. Ask a question for each quiz and cycle";
-        await textbox.fill(message);
-        await expect(textbox).toContainText(message);
-
-        // Temporary fix: The test goes quicker than a real user and submits before the demo status has loaded
-        // This means that a demo modal would be shown when submitting
-        await page.waitForTimeout(500);
-
-        setFixture("roman-britain-1");
-        await sendMessage.click();
-      });
-
-      await test.step("Iterate through the fixtures", async () => {
-        await page.waitForURL(/\/aila\/.+/);
-        await waitForGeneration(page, generationTimeout);
-        await expectSectionsComplete(page, 1);
-
-        setFixture("roman-britain-2");
-        await continueChat(page);
-        await waitForGeneration(page, generationTimeout);
-        await expectSectionsComplete(page, 3);
-
-        setFixture("roman-britain-3");
-        await continueChat(page);
-        await waitForGeneration(page, generationTimeout);
-        await expectSectionsComplete(page, 7);
-
-        setFixture("roman-britain-4");
-        await continueChat(page);
-        await waitForGeneration(page, generationTimeout);
-        await expectSectionsComplete(page, 10);
-
-        setFixture("roman-britain-5");
-        await continueChat(page);
-        await waitForGeneration(page, generationTimeout);
-        await expectSectionsComplete(page, 10);
-
-        await expectFinished(page);
-      });
-    },
-  );
-});
+test(
+  "Full aila flow with Romans fixture",
+  { tag: "@common-auth" },
+  async ({ page }, testInfo) => {
+    const generationTimeout = FIXTURE_MODE === "record" ? 75000 : 50000;
+    test.setTimeout(generationTimeout * 5);
+
+    // The chat UI has a race condition when you submit a message too quickly after the previous response
+    // This is a temporary fix to fix test flake
+    async function letUiSettle() {
+      return await page.waitForTimeout(testInfo.retry === 0 ? 500 : 6000);
+    }
```

Comment:
This is the new solution to the problem. If the last test run failed, wait 6 seconds before clicking `continue` again

This means that we'll get a quick test run if the code under test works well, and if it doesn't we'll see that there was a retry in the test summary